### ISSUE TITLE
Sync BlackHole clock with other devices

### ISSUE THREAD

**Issue by xavrr** [ExistentialAudio/BlackHole#27]
Hello, I'm using blackhole on catalina. (Many thanks for this great work !)

I'm using blackhole to record audio during many hours and I've noticed that it seems to introduce a delay that will add up to multiple seconds after couple of hours.
Changing the audio frequency and back using audio midi setup seems to "reset" the latency.

Does anyone have experienced the same issue ?

**Comment by devinroth** [comment]
First off BlackHole doesn’t currently support syncing clocks with other devices. After a while you’ll notice a delay as the clock get further out of sync. What programs are you using with BlackHole? If you are using a multi output or aggregate make sure drift correction is enabled. If the audio software you are using does support drift correction this will unfortunately continue. 

Clock syncing is a very important issue for me. My goal is to allow users to select an audio device to use as a clock source.

**Comment by xavrr** [comment]
Thanks for your answer.
My setup is  python midi event generator>garageband -> blackhole -> python audio recorder.
I'm recording both midi events and audio, and I really need it to be synced.
Following your message, II'm trying to use an aggregated device (mac microphone+blackhole) using the mac microphone as clock source and checking drift correction on blackhole. 
Does it have a chance of making a change ?

**Comment by devinroth** [comment]
It should work in theory with drift correction but aggregate devices have been known to be sketchy. Ultimately clock syncing is the solution but I haven’t had time to dig into that yet. Let me know the result.

**Comment by ivanvfer** [comment]
Hi, first of all many thanks for this cool development! 

I use Tonebridge to support different guitar effects, and with BlackHole I can route the guitar sound to Reaper (using BlackHole channels other than 1-2). With Reaper, I can mix it with the mic input, and route the mixed audio to Skype using the first two channels of BlackHole.

I have also noticed that, after playing for some time, there was a small delay in the guitar sound. I used an aggregate device in Reaper, with BlackHole and a sound card. Enabling drift correction, with the sound card as the clock source, seems to have addressed the delay issue (at least, guitar delay is not noticeable after ~2h).
